code/modules/reagents/chems/chems_painkillers.dm

```diff
@@ -25,37 +34,76 @@
 
 	M.add_chemical_effect(CE_PAINKILLER, (pain_power * effectiveness))
 
-	if(dose > 0.5 * overdose)
-		M.add_chemical_effect(CE_SLOWDOWN, 1)
-		if(prob(1))
-			SET_STATUS_MAX(M, STAT_SLUR, 10)
-	if(dose > 0.75 * overdose)
-		M.add_chemical_effect(CE_SLOWDOWN, 1)
-		if(prob(5))
-			SET_STATUS_MAX(M, STAT_SLUR, 20)
-	if(dose > overdose)
-		M.add_chemical_effect(CE_SLOWDOWN, 1)
-		SET_STATUS_MAX(M, STAT_SLUR, 30)
-		if(prob(1))
-			SET_STATUS_MAX(M, STAT_WEAK, 2)
-			SET_STATUS_MAX(M, STAT_DROWSY, 5)
+	if(!narcotic)
+		return
+	if(dose > 0.5 * additional_effect_threshold)
+		M.add_chemical_effect(CE_SLOWDOWN, slowdown_severity)
+		if(prob(15))
+			SET_STATUS_MAX(M, STAT_SLUR, (slur_severity * 10))
+
+	if(dose > 0.75 * additional_effect_threshold) //minor side effects may start here
+		M.add_chemical_effect(CE_SLOWDOWN, slowdown_severity)
+		if(prob(30))
+			SET_STATUS_MAX(M, STAT_SLUR, (slur_severity * 20))
```

Comment:
It may be worth doing `if(prob(30) && slur_severity)` for each of these status checks, to avoid calling `SET_STATUS_MAX(M, STAT_SLUR, 0)` for some chems. The status set macro is significantly slower than a truthiness check and you're doing a bunch of them in this proc potentially.